pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -127,34 +129,129 @@ public String getStatusDescription() {
       return statusDescription.toString();
     }
   }
+  
+  public static abstract class BaseServiceStatusCallback implements ServiceStatusCallback {
```

Comment:
I don't think we need this base class. ServiceStatusCallback does not always associate with Helix cluster info (e.g. MultipleCallbackServiceStatusCallback), also the methods are not very commonly used. I don't like the way of returning statusDescription by setting a member variable from the parent class, it's not very readable. The base class IdealStateMatchServiceStatusCallback is already good enough IMO.

Reply:
RealtimeConsumptionCatchupServiceCallback doesn't really belong under IdealStateMatchServiceStatusCallback. What is the suggestion? Keep RealtimeConsumptionCatchupServiceCallback independent and directly implement ServiceStatusCallback interface?

Reply:
Yes, directly implement ServiceStatusCallback